is_core/auth_token/middleware.py

```diff
@@ -26,23 +26,42 @@ def process_request(self, request):
         request.user = SimpleLazyObject(lambda: get_user(request))
         request._dont_enforce_csrf_checks = utils.dont_enforce_csrf_checks(request)
 
+    def update_token_and_cookie(self, request, response, max_age, expires):
+        request.token.save()
+        response.set_cookie(settings.AUTH_COOKIE_NAME, force_text(request.token.key), max_age=max_age,
+                            expires=expires, httponly=settings.AUTH_COOKIE_HTTPONLY,
+                            secure=settings.AUTH_COOKIE_SECURE, domain=settings.AUTH_COOKIE_DOMAIN)
+        return response
+
+    def set_auth_expiration_header(self, request, response):
+        response['HTTP_X_AUTHENTICATION_EXPIRATION'] = request.token.time_to_expiration
+
+    def is_token_renewal_exempt(self, request):
+        return True if getattr(request, 'auth_token_renewal_exempt', False) or request.META.get(
+            'HTTP_X_AUTH_TOKEN_RENEWAL_EXEMPT', None) else False
```

Comment:
if true pattern :)

`return getattr(request, 'auth_token_renewal_exempt', False) or 'HTTP_X_AUTH_TOKEN_RENEWAL_EXEMPT' in request.META`